rust/agama-cli/src/profile.rs

```diff
@@ -76,123 +70,245 @@ pub enum ProfileCommands {
         /// AutoYaST specific ones. Supported files are json, jsonnet, sh for Agama profiles and ERB, XML, and rules/classes directories
         /// for AutoYaST support.
         url: String,
-        /// Specific directory where all processing happens. By default it uses a temporary directory
+        /// Ignored, formerly the directory for intermadiate files.
         dir: Option<PathBuf>,
     },
 }
 
-fn validate(path: &PathBuf) -> anyhow::Result<()> {
-    let validator = ProfileValidator::default_schema()?;
-    let result = validator
-        .validate_file(path)
-        .context(format!("Could not validate the profile {:?}", path))?;
-    match result {
-        ValidationResult::Valid => {
-            println!("{} The profile is valid.", style("\u{2713}").bold().green(),);
+/// TODO better name
+// Represents the ways user can specify the input on the command line
+// and passes appropriate representations to the web API
+enum CliInput {
+    Url(String),
+    Path(String),
+    Stdin,
+    // the full text as string
+    Full(String),
+}
+
+impl CliInput {
+    fn new(url_or_path: String) -> Self {
```

Comment:
convention is to call such conversion methods `from_string` https://doc.rust-lang.org/1.0.0/style/style/naming/README.html#general-conventions-[rfc-#430]